packages/bsky/src/api/app/bsky/actor/getProfile.ts

```diff
@@ -5,10 +5,12 @@ import AppContext from '../../../../context'
 
 export default function (server: Server, ctx: AppContext) {
   server.app.bsky.actor.getProfile({
-    auth: ctx.authOptionalVerifier,
+    auth: ctx.authOptionalAccessOrRoleVerifier,
     handler: async ({ auth, params }) => {
       const { actor } = params
-      const requester = auth.credentials.did
+      const requester = 'did' in auth.credentials ? auth.credentials.did : null
+      const canViewTakendownProfile =
+        'triage' in auth.credentials && auth.credentials.triage
```

Comment:
I think we can use the `type` field on the credentials now for these checks:
```suggestion
      const requester = auth.credentials.type === 'access' ? auth.credentials.did : null
      const canViewTakendownProfile = auth.credentials.type === 'role'
```

Reply:
hmm... I was thinking that at some point, we may only want to give access to these content to mods with specific level of permission in which case, we would want to check for specific level of access which is why I used this approach.